app/src/main/java/com/example/taskyy/ui/viewmodels/ReminderViewModel.kt

```diff
@@ -77,34 +99,106 @@ class ReminderViewModel @Inject constructor(
         }
     }
 
-    private fun formatTimeSelected(time: String) {
-        var timeString = time
-        if (timeString == "0:0") {
-            timeString = ("00:00")
-        }
-        if (timeString == "12:0") {
-            timeString = ("12:00")
+    @OptIn(ExperimentalMaterial3Api::class)
+    private fun formatTimeSelected(time: TimePickerState) {
+
+        val localTime = org.joda.time.LocalTime(time.hour, time.minute)
+        val twelveHourDateFormat =
+            DateTimeFormat.forPattern("hh:mm aa") // 12-hour format with AM/PM
+        val formattedTime = twelveHourDateFormat.print(localTime)
+
+        _state.update { it.copy(selectedTime = formattedTime) }
+    }
+
+    private fun save(title: String, description: String) {
+        val newReminder = Reminder(
+            alarmType = _state.value.formattedReminderType,
+            title = title,
+            description = description,
+            timeInMillis = _state.value.formattedReminderTime,
+            id = Random.nextInt(1..10000),
+
+            )
+        viewModelScope.launch {
+            when (val save = agendaRepository.saveReminder(newReminder)) {
+                is Result.Success -> {
+                    eventChannel.send(ReminderEvent.SaveSuccessful)
+                }
+
+                is Result.Error -> {
+                    eventChannel.send(ReminderEvent.SaveFailed(save.error.asUiText()))
+                }
+            }
         }
-        val twelveHourDateFormat = DateTimeFormatter.ofPattern("HH:mm", Locale.ENGLISH)
-        val time24 = LocalTime.parse(timeString, twelveHourDateFormat)
-        val formatter12 = DateTimeFormatter.ofPattern("hh:mm a")
-        val time12 = time24.format(formatter12)
+    }
+
+    private fun formatTimeForDB(selectedTime: String, selectedDate: String) {
```

Comment:
I wouldn't create my own date format for the DB. There are standardized formats every date formatter can understand. That's typically either storing the Long timestamp or an ISO date string (like `2011-10-05T14:48:00.000Z`)

Reply:
Well the date and time are seperate strings so I have to convert them both to milliseconds separately and then add them together to get the date and time in milliseconds. I'm not sure how else I would do it

Reply:
You can work with one single `LocalDateTime` state which contains both date and time information. You can then format this instance once as HH:mm and once for the date

Reply:
How would that work? I think I understand what your saying but I can't visualize what the implementation would look like

Reply:
Something like this:
```
data class MyScreenState(
    val dateTime: LocalDateTime = LocalDateTime.now(),
    ...
)
```
And in your UI:
```
val formattedDate = remember(state.dateTime) {
    DateTimeFormatter.ofPattern("dd MM yyyy").format(state.dateTime)
}
val formattedDate = remember(state.dateTime) {
    DateTimeFormatter.ofPattern("HH mm").format(state.dateTime)
}
```

Reply:
And when you select a `LocalDate` from the UI and want to set that in your LocalDateTime instance of the state, you should be able to do this:
```
state = state.copy(dateTime = state.dateTime.with(newChosenDate)
```